### PR TITLE
Fix typos discovered by codespell

### PR DESCRIPTION
**Pull request by cclauss**
## References
_Add references/links to any related issues or PRs. These may include:_
* Fixes #issue-number (if this fixes an issue ticket)
* Related to DSpace/RestContract#pr-number  (if a corresponding REST Contract PR exists)

## Description
Short summary of changes (1-2 sentences).
Fix typos discovered by codespell -- https://github.com/codespell-project/codespell

```
codespell --skip="*.json,*.pdf,*.rtf,*.sql,*.xml,*.xsl" --write-changes \
  --ignore-words-list=ane,bais,bui,bund,bunds,corus,currenty,datas,errorprone,fo,hald,laf,ment,myu,nd,noes,ois,ore,ot,rin,statics,te,theses,ue,usera,wee,wel 
```

## Instructions for Reviewers
Please add a more detailed description of the changes made by your PR. At a minimum, providing a bulleted list of changes in your PR is helpful to reviewers.

List of changes in this PR:
* First, ...
* Second, ...

**Include guidance for how to test or review your PR.** This may include: steps to reproduce a bug, screenshots or description of a new feature, or reasons behind specific changes. 

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome).
However, reviewers may request that you complete any actions in this list if you have not done so. If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_

- [x] My PR is **created against the `main` branch** of code (unless it is a backport or is fixing an issue specific to an older branch).
- [ ] My PR is **small in size** (e.g. less than 1,000 lines of code, not including comments & integration tests). Exceptions may be made if previously agreed upon.
- [ ] My PR **passes Checkstyle** validation based on the [Code Style Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Style+Guide).
- [ ] My PR **includes Javadoc** for _all new (or modified) public methods and classes_. It also includes Javadoc for large or complex private methods.
- [ ] My PR **passes all tests and includes new/updated Unit or Integration Tests** based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [ ] My PR **includes details on how to test it**. I've provided clear instructions to reviewers on how to successfully test this fix or feature.
- [ ] If my PR includes new libraries/dependencies (in any `pom.xml`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [ ] If my PR modifies REST API endpoints, I've opened a separate [REST Contract](https://github.com/DSpace/RestContract/blob/main/README.md) PR related to this change.
- [ ] If my PR includes new configurations, I've provided basic technical documentation in the PR itself.
- [ ] If my PR fixes an issue ticket, I've [linked them together](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

### CODE REVIEW

**Review by kshepherd — CHANGES_REQUESTED** [review]
Thanks @cclauss almost everything here looks perfect.

I made a note of the only non-comment, non-string-output (log msgs etc) changes that actually affected variable names, and it looks OK and consistent so as not to introduce problems:

```
dspace-api/src/main/java/org/dspace/content/crosswalk/SubscriptionDsoMetadataForEmailCompose.java
dspace-api/src/main/java/org/dspace/importer/external/metadatamapping/contributor/EnhancedSimpleMetadataContributor.java
dspace-api/src/test/java/org/dspace/content/virtual/ConcatenateTest.java
dspace-server-webapp/src/main/java/org/dspace/app/rest/hdlresolver/HdlResolverRestController.java
```

The only thing I noticed in my scan, was that this configuration property is updated in code / tests: `core.authorization.community-admin.item.reinstatiate` (to 'reinstantiate') but not in `dspace/config/dspace.cfg` where it is commented out as a default. This made me wonder if some other config property changes could also be affected, so I'd suggest we either include dspace.cfg in your codespell analysis input, or at least identify all changed config property keys and check .cfg files for usage of those properties.

Thanks!

**Review by hardyoyo — COMMENTED** [review]
This looks great, I spy at least one actual bug fix in this typo fix. I am an emeritus committer now, so I will leave the actual approval to someone else, but I am +1 on this, and am inspired by this contribution, thanks for making it!

File: dspace/config/dspace.cfg
Comment (saschaszott) [inline dspace/config/dspace.cfg:353]: This configuration key `core.authorization.community-admin.item.reinstantiate` is referenced in `AuthorizeConfiguration::canCommunityAdminPerformItemReinstatiate` - some DS instances could have been copied the key into `local.cfg`. If the key name is changed (corrected), a note needs to be added to the release notes that indicate this renaming.

File: dspace/config/dspace.cfg
Comment (saschaszott) [inline dspace/config/dspace.cfg:370]: This configuration key core.authorization.collection-admin.item.reinstantiate is referenced in AuthorizeConfiguration::canCollectionAdminPerformItemReinstatiate - some DS instances could have been copied the key into local.cfg. If the key name is changed (corrected), a note needs to be added to the release notes that indicate this renaming.

File: dspace/config/dspace.cfg
Comment (saschaszott) [reply dspace/config/dspace.cfg:353]: BTW: What is the correct term `reinstate` (more than 100 matches in the Java code base) or `reinstantiate` (currently, only 1 match)?

Comment (saschaszott) [reply dspace/config/dspace.cfg:353]: BTW 2: Why is codespell complaining about `reinstatiate` in a string literal (in `AuthorizeConfiguration::canCommunityAdminPerformItemReinstatiate`) but not in a method name (`canCommunityAdminPerformItemReinstatiate`)? The same observation holds for `canCollectionAdminPerformItemReinstatiate`.

Comment (kshepherd) [reply dspace/config/dspace.cfg:353]: The action in our documentation and other controls is "reinstate", I believe - it could be correct to call this 'an act of reinstantiation' but you raise a good point. I think we should change all instances of this to reinstate, and add a note about the configuration property change to the release notes / doco

**Comment by cclauss** [comment]
Let's do this...  I will revert all changes related to `reinstatiate` and then we can migrate that discussion to another PR.

**Comment by cclauss** [comment]
Re-added `reinstatiate`
```
codespell --skip="*.json,*.pdf,*.rtf,*.sql,*.xml,*.xsl" --write-change \
  --ignore-words-list=ane,bais,bui,bund,bunds,corus,currenty,datas,errorprone,fo,hald,laf,ment,myu,nd,noes,ois,ore,ot,reinstatiate,rin,statics,te,theses,ue,usera,wee,wel
```

**Review by kshepherd — APPROVED** [review]
+1 the only remaining code changes are with local variables named `splitted` -> `split` and they work well. Everything else is comment or log/message output.

Thanks for this really useful cleanup task @cclauss , it will help keep all our code and configs more readable and maintainable.

Merging now as this only needs 1 approval, has other +1s and I have checked the code changes by hand.

**Comment by cclauss** [comment]
* #9757

File: dspace/config/dspace.cfg
Comment (mwoodiupui) [reply dspace/config/dspace.cfg]: "Reinstate" seems correct, since it seems to be paired with "withdraw".  "Reinstantiate" ought to mean to create again from a type or template, and seems unlikely here.

**Comment by tdonohue** [comment]
@kshepherd : Could I ask why you flagged this as `needs documentation`?  I ask because, at a glance, it's unclear to me how to document the typo fixes in this PR.  Likely, we should describe the documentation needs in a comment on this PR, so that we know how to appropriately fix the 9.0 docs.

**Comment by kshepherd** [comment]
> @kshepherd : Could I ask why you flagged this as `needs documentation`? I ask because, at a glance, it's unclear to me how to document the typo fixes in this PR. Likely, we should describe the documentation needs in a comment on this PR, so that we know how to appropriately fix the 9.0 docs.

@tdonohue ah, that was added when there were some configuration properties changed, but they were not included in this PR in the end, we discussed addressing that in a separate issue/pr (see #9757). I've removed it now, thanks

**Pull request merged**
